pkg/services/sqlstore/migrations/folder_mig.go

```diff
@@ -0,0 +1,36 @@
+package migrations
+
+import (
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addFolderMigrations(mg *migrator.Migrator) {
+	mg.AddMigration("create folder table", migrator.NewAddTableMigration(folderv1()))
+
+	// copy any existing folders in the dashboard table into the new folder
+	// table. The *legacy* parent folder ID, stored as folder_id  in the
+	// dashboard table, is always going to be "0" so it is safe to convert to a parent UID.
+	mg.AddMigration("copy existing folders from dashboard table", migrator.NewRawSQLMigration(
+		"INSERT INTO folder (id, uid, org_id, title, parent_uid, created, updated) SELECT id, uid, org_id, title, folder_id, created, updated FROM dashboard WHERE is_folder = 1;",
+	).Postgres("INSERT INTO folder (id, uid, org_id, title, parent_uid, created, updated) SELECT id, uid, org_id, title, folder_id, created, updated FROM dashboard WHERE is_folder = true;"))
```

Comment:
I think this is where `folder_id` should be replaced with an empty string, and so in the migration statement above

Reply:
Works for me! All I really needed was the current id `0` cast to a string `"0"`.  

(`0` is the "General" folder ID, so the parent uid should be `"0"` and not `""`)

Reply:
If it works - great, but I suspect one can make a folder/dashboard with UID "0". I think empty string means the root folder.

Reply:
OH HEY! I just found this interesting definition while poking around the existing folder IDs:

https://github.com/grafana/grafana/blob/c27aac0d385bfe695dec5f1b268940b4a19d3c50/pkg/services/accesscontrol/models.go#L279

So I will add `folders.RootFolderUID` const ("general") and use that so we're consistent. I'd appreciate help updating the migration; I'm not sure how to do that efficiently (instead of "insert from dashboards then update every record to have "general" as the UID").

Reply:
I would suggest setting the default for `parent_uid` column to the `GeneralFolderUID` so that we don't need to handle this in the migration WDYT?

Reply:
nit; I would also suggest moving the `GeneralFolderUID` constant under the `folder` service.

Reply:
That makes great sense @papagian thank you!
and yes, as i said in the comment I'm adding `folders.RootFolderUID` and not using that one.